src/test/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+set(BINARY Test${CMAKE_PROJECT_NAME})
+
+find_package( Threads )
+
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+endif()
+execute_process(COMMAND ${CMAKE_COMMAND} --build .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "Build step for googletest failed: ${result}")
+endif()
+
+# Prevent overriding the parent project's compiler/linker
+# settings on Windows
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+# Add googletest directly to our build. This defines
+# the gtest and gtest_main targets.
+
+add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
+                 ${CMAKE_CURRENT_BINARY_DIR}/googletest-build
+                 EXCLUDE_FROM_ALL)
+
+set_target_properties( gtest PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} )
+set_target_properties( gtest_main PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} )
+
+add_executable(${BINARY}
+	TestMain.cpp
+    TestDASHTree.cpp
+	TestHLSTree.cpp
+	TestHelper.cpp
+    ../parser/DASHTree.cpp
+	../parser/HLSTree.cpp
+    ../parser/PRProtectionParser.cpp
+    ../common/AdaptiveTree.cpp
+    ../helpers.cpp
+    ../oscompat.cpp
+    )
+
+target_link_libraries(${BINARY} PUBLIC ${CMAKE_PROJECT_NAME} gtest_main gtest)
```

Comment:
Please add `${CMAKE_DL_LIBS}` in this line to fix cross-compiling.

Reply:
Please add `${CMAKE_DL_LIBS}` or else this still won't cross-compile. Unless this breaks native builds?